spec/controllers/local_authorities_controller_spec.rb

```diff
@@ -63,18 +63,73 @@
   end
 
   describe "POST upload_links_csv" do
-    before { @local_authority = create(:local_authority) }
-    let(:path) { Rails.root.join("spec/lib/local-links-manager/import/fixtures/imported_links.csv") }
-    let(:csv) { Rack::Test::UploadedFile.new(path, "text/csv", true) }
-    let(:url_regex) { /http:\/\/.+\/local_authorities\/#{@local_authority.slug}/ }
+    context "with a valid CSV" do
+      before { @local_authority = create(:local_authority) }
+      let(:path) { Rails.root.join("spec/lib/local-links-manager/import/fixtures/imported_links.csv") }
+      let(:csv) { Rack::Test::UploadedFile.new(path, "text/csv", true) }
+      let(:url_regex) { /http:\/\/.+\/local_authorities\/#{@local_authority.slug}/ }
 
-    it "retrieves HTTP found" do
-      login_as_stub_user
-      post(:upload_links_csv, params: { local_authority_slug: @local_authority.slug, csv: })
+      it "retrieves HTTP found" do
+        login_as_stub_user
+        post(:upload_links_csv, params: { local_authority_slug: @local_authority.slug, csv: })
+
+        expect(response.status).to eq(302)
+        expect(response.location).to match(url_regex)
+        expect(response.headers["Content-Type"]).to eq("text/html; charset=utf-8")
+      end
+    end
 
-      expect(response.status).to eq(302)
-      expect(response.location).to match(url_regex)
-      expect(response.headers["Content-Type"]).to eq("text/html; charset=utf-8")
+    context "with errors in the CSV" do
+      before do
+        @local_authority = create(:local_authority)
```

Comment:
It is a bit strange we're using instance variables vs let in RSpec files, I can see you're following the pattern in the file but maybe you can change all the instance vars to `let`?

Reply:
Agreed, made into an instance var.